ocaml/libs/xapi-compression/xapi_gzip.ml

```diff
@@ -0,0 +1,67 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+(* This is a minimal test CLI to exercise gzip for compression and
+   uncompression using the Gzip module; which in turn spawns a process via
+   Forkexecd *)
+
+module C = Cmdliner
+
+module Gzip = Xapi_compression.Make (struct let executable = "/bin/gzip" end)
+
+let help =
+  [
+    `P "These options are common to all commands."
+  ; `S "MORE HELP"
+  ; `P "Use `$(mname) $(i,COMMAND) --help' for help on a single command."
+  ]
+
+(** copy bytes from file descriptor [src] to [dst] *)
+let copy ~src ~dst =
+  let size = 1024 in
```

Comment:
for OCaml I think a 64k buffer size is optimal (anything above that causes the Unix module to limit it anyway to 64k: https://github.com/ocaml/ocaml/blob/trunk/otherlibs/unix/write_unix.c#L76 
https://github.com/ocaml/ocaml/blob/3f459df2b6dc2224697b5b37ec82ca1df40a1e5a/otherlibs/unix/unixsupport.h#L95
)